hw/chisel/src/test/scala/snax/csr_manager/CsrManagerWithReadOnlyTest.scala

```diff
@@ -0,0 +1,69 @@
+package snax.csr_manager
+
+import chisel3._
+import org.scalatest.flatspec.AnyFlatSpec
+import chiseltest._
+import scala.math.BigInt
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.Tag
+import scala.util.control.Breaks._
+
+class CsrManagerWithReadOnlyTest
+    extends AnyFlatSpec
+    with ChiselScalatestTester
+    with Matchers
+    with HasCsrManagerTest {
+
+  "DUT" should "pass" in {
+    test(
+      new CsrManagerWithReadOnly(
+        CsrManagerWithReadOnlyTestParameters.csrNum,
+        CsrManagerWithReadOnlyTestParameters.csrNumReadOnly,
+        CsrManagerWithReadOnlyTestParameters.csrAddrWidth
+      )
+    ).withAnnotations(Seq(WriteVcdAnnotation)) { dut =>
+      base_csr_manager_test(dut)
+
+      // ***********************************************************************
+      // test read only csr
+      // ***********************************************************************
+
+      // drive the read only csr
+      dut.io.read_only_csr(0).poke(1.U)
+      dut.io.read_only_csr(1).poke(2.U)
```

Comment:
hehe is there a way to do something like putting random values?

```scala
dut.io.read_only_csr(0).poke(random)
dut.io.read_only_csr(1).poke(random)
```

Reply:
Test specific data for control and random data for compute